index.js

```diff
@@ -129,7 +239,26 @@ function filterByGenre() {}
       }
     ];
  */
-function getAllMoviesReleasedAtOrBeforeYear() {}
+function getAllMoviesReleasedAtOrBeforeYear(movies,year) {
+let result = [];
+if (movies.length === 0){
+  return result;
+}
+
+for (i = 0; i < movies.length; i++){
+  let movieYear = movies[i].released;//get the string of the movie release date
+  let newYear = movieYear.substr(movieYear.length-4,movieYear.length);//cut the string so I can get the year part only
```

Comment:
I'm a big fan of the `.substr()` call here but I think we can find a better variable names than `newYear` and `movieYear`.

Reply:
I see that make sense